benches/rotation.rs

```diff
@@ -0,0 +1,82 @@
+#![feature(duration_constants)]
+
+use std::time::{Duration, Instant};
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use lazy_static::lazy_static;
+use rayon::prelude::*;
+
+const K: u64 = 1024;
+const M: u64 = K * K;
+const FILE_SIZE: u64 = 10 * M;
+const FILE_COUNT: u32 = 10;
+
+lazy_static! {
+    static ref HANDLE: log4rs::Handle =
+        log4rs::init_config(mk_config(FILE_SIZE, FILE_COUNT)).unwrap();
+    static ref MSG: String = "0".repeat(80 * M as usize);
+}
+
+fn log_group(c: &mut Criterion) {
+    lazy_static::initialize(&HANDLE);
+
+    let mut group = c.benchmark_group("Log");
+    group.sample_size(10);
+    group.measurement_time(Duration::SECOND * 30);
+    group.bench_function("log", |b| b.iter(a::write_log));
+}
+
+fn log2_group(c: &mut Criterion) {
+    lazy_static::initialize(&HANDLE);
+
+    let mut group = c.benchmark_group("Parallel Log");
+    group.sample_size(10);
+    group.measurement_time(Duration::SECOND * 30);
+    group.bench_function("log", |b| {
+        b.iter_custom(|iter| {
+            let now = Instant::now();
+            (0..iter).into_par_iter().for_each(|_| a::write_log());
+            now.elapsed()
+        })
+    });
+}
+
+criterion_group!(benches, log_group, log2_group);
```

Comment:
So this is a minor thing, but do we want to consider cleaning up logs after each run? I ask for two reasons, the first being that cleaning up after the tests might be nice. The second is that whichever test runs first will create the files as it rolls them, and the second test will have to create new log files and roll the existing ones over. 

Similarly do we want the sample size to equal to the file count?

Reply:
Absolutely, so far this whole config has been stolen from @yakov-bakhmatov 's bench suite. I'm going to use tempdir crate to clean those up automatically. Also in general this bench stuff is major WIP